content/getting-started/connect-the-pim-4x/step-2.md

```diff
@@ -0,0 +1,120 @@
+
+### 6- Ochestrate your global connector
+
+A crucial step. In most cases, you will need to manually map some PIM information with the data sent or received in the software you want to connect.
+
+#### In this tutorial
+_For our example, here is the global orchestration of our connector. First, every evening, you will extract the list of products that were created/updated during the day, in an XML file. Then, this file is read by your connector. ERP products are transformed into PIM products, by applying a mapping of the ERP information with the right attributes of the PIM. To finish, the connector sends the PIM products by using the [patch](/api-reference.html#patch_products) endpoint of our API._
+
+## Some pro tips
+
+As we want you to build the best connector ever, here are some tips to help you cook it the right way.
+
+### Start small
+The use case we took here is really simple, as you can see. In real life, connections maybe a bit more tricky with a lot of different use cases to handle.
+
+We highly recommend you to **start small**. Focus on one single use case at first. Then, when the first one is robust and approved, add some more.
+
+Don't try to do everything at once, they are high chances you won't see the end of it and your connector won't be usable.
+
+### UI or not UI?
+
+Some connectors also require the developement of a UI. For example, it can be very useful in the case you want your final users to be able to input some configuration. Indeed, in some cases, you may want to ask them for product, channel or locale selections.
+
+Another option is to provide a configuration file to input this configuration. It's usually faster to develop but keep in mind that you will be far from delivering the best UX for your final users. :wink:
+
+### Think scalability and performance
+
+It is really important that you take into consideration the scalability and performance of your connector.
+
+As stated previously, again, we advise you to start small. Benchmark your connector with a small volume, then test it with a real-life catalog volume.
+
+To help you in this task, here are examples of questions you can ask yourself:
+- _"How many products, assets, reference entities or options your connector will handle?"_
+- _"How big are your products, assets or reference entities in termes of number of attributes completed?"_
+
+Also, don't forget to think about the fact that every API call you are doing has an impact on the **PIM load**. Always keep in mind that your connector won't be the only one connected to the PIM. Try to have a load print as low as possible. To do so, you can use caches for example.
+
+### Test, test and test
+
+Testing should not be optional. You will need to test your full use case and also focus on all the edge cases that can happen.
+
+### Use the PHP client
+
+If you want to use PHP to build your connector, we highly recommend the use of our PHP client.
```

Comment:
A link to our client would be awesome here! thanks.

Reply:
```suggestion
If you want to use PHP to build your connector, we highly recommend the use of [our PHP client](/php-client/introduction.html).
```